### PR TITLE
Commit regfile prep

### PR DESCRIPTION
**Pull request by cgwalters**
Depends: https://github.com/ostreedev/ostree/pull/1256

### CODE REVIEW

**Comment by cgwalters** [comment]
Rebased 🏄

File: src/libostree/ostree-core.c [hunk 1]
```diff
@@ -333,119 +340,72 @@ _ostree_zlib_file_header_new (GFileInfo         *file_info,
...
+  /* Write NULs for alignment. At the moment this is a constant 4 bytes (i.e.
+   * align to 8, since the length is 4 bytes). For now, I decided to keep some
+   * of the (now legacy) more generic logic here in case we want to revive it
+   * later.
+   */
+  const guint alignment = 8;
+  const guchar padding_nuls[8] = {0, 0, 0, 0, 0, 0, 0, 0};
+  guint bits;
+  if (alignment == 8)
+    bits = alignment_offset & 7;
```
Comment (jlebon) [inline src/libostree/ostree-core.c]: Not new to this patch, though can we add a comment explaining these? This is essentially a more efficient way to do modulo 8 and 4, right?

File: src/libostree/ostree-core.c [hunk 2]
```diff
@@ -283,247 +285,142 @@ ostree_validate_collection_id (const char *collection_id, GError **error)
...
-  ret_bytes_written += bytes_written;
-
-  bytes_written = 0;
-  if (!ot_gio_write_update_checksum (output, g_variant_get_data (variant),
-                                     variant_size, &bytes_written, checksum,
-                                     cancellable, error))
-    return FALSE;
-  ret_bytes_written += bytes_written;
+  g_string_append_len (buf, (char*)&variant_size_u32_be, sizeof (variant_size_u32_be));
+  const gsize alignment_offset = sizeof(variant_size_u32_be);
```
Comment (jlebon) [inline src/libostree/ostree-core.c]: Minor nit: `sizeof` spacing.

**Comment by cgwalters** [comment]
Rebased 🏄 and fixups ⬆️

**Comment by jlebon** [comment]
@rh-atomic-bot r+ d2c654f

**Pull request closed without merging**